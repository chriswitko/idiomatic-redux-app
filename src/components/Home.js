import React, { Component } from 'react';
import Siema from 'siema';
import Nav from './Nav';

class Home extends Component {
  constructor(props={}) {
    super();

    this.prev = this.prev.bind(this)
    this.next = this.next.bind(this)

    this.state = props;
  }
  // getInitialState() {
  //   return {data: []};
  // }

  componentDidMount() {
    this.siema = new Siema();
  }

  prev() {
    this.siema.prev()
  };
  
  next() {
    this.siema.next()
  };  

  render() {
    return (
      <div>
        <Nav />
        <div className="siema">
          <div><img src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" alt="Siema image" /></div>
          <div><img src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" alt="Siema image" /></div>
          <div><img src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" alt="Siema image" /></div>
          <div><img src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" alt="Siema image" /></div>
        </div>
        <button onClick={this.prev}>Prev</button>
        <button onClick={this.next}>Next</button>
      </div>
    );
  }  
}

export default Home;
