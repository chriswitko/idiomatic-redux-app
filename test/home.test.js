import Home from '../src/components/Home';
// import React from 'react';

describe('Home', () => {
  it('should render a Home component', () => {
      const wrapper = shallow(
          <Home />
      );
      expect(wrapper).toMatchSnapshot();
  });

  it('should render a small label', () => {
      const wrapper = shallow(
          <Label small>Hello Jest!</Label>
      );
      expect(wrapper).toMatchSnapshot();
  });

  it('should render a grayish label', () => {
      const wrapper = shallow(
          <Label light>Hello Jest!</Label>
      );
      expect(wrapper).toMatchSnapshot();
  });
})

