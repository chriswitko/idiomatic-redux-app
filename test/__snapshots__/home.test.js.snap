exports[`test should render a Home component 1`] = `
<div>
  <Nav />
  <div
    className="siema">
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" />
    </div>
  </div>
  <button
    onClick={[Function]}>
    Prev
  </button>
  <button
    onClick={[Function]}>
    Next
  </button>
</div>
`;

exports[`test should render a label 1`] = `
<div>
  <Nav />
  <div
    className="siema">
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--pink.svg" />
    </div>
    <div>
      <img
        alt="Siema image"
        src="https://pawelgrzybek.com/siema/assets/siema--yellow.svg" />
    </div>
  </div>
  <button
    onClick={[Function]}>
    Prev
  </button>
  <button
    onClick={[Function]}>
    Next
  </button>
</div>
`;
